solidity/test/FullyBackedECDSAKeepFactoryTest.js

```diff
@@ -0,0 +1,1633 @@
+      assert.isFalse(
+        await keepFactory.isOperatorUpToDate(members[0], application)
+      )
+    })
+
+    it("returns true if the operator bonding value increase is below weight precision", async () => {
+      const unbondedValue = minimumBondableValue.muln(10) // (20 * 10^18) * 10
+      await setUnbondedValue(members[0], unbondedValue)
+
+      await keepFactory.registerMemberCandidate(application, {
+        from: members[0],
+      })
+
+      await setUnbondedValue(members[0], unbondedValue.add(precision).subn(1)) // ((20 * 10^18) * 10) + 10^18 - 1
+
+      // Precision (pool weight divisor) is 10^18
+      // (20 * 10^18 * 10) / 10^18                 =  200
+      // ((20 * 10^18 * 10) + (10^18 - 1)) / 10^18 =~ 200.99
+      // Ethereum uint256 division performs implicit floor:
+      // The weight dit not change: 200 == floor(200.99)
+      assert.isTrue(
+        await keepFactory.isOperatorUpToDate(members[0], application)
+      )
+    })
+
+    it("returns false if the operator stake dropped below minimum", async () => {
+      await keepFactory.registerMemberCandidate(application, {
+        from: members[0],
+      })
+
+      await setUnbondedValue(members[0], minimumBondableValue.subn(1))
+
+      assert.isFalse(
+        await keepFactory.isOperatorUpToDate(members[0], application)
+      )
+    })
+
+    it("reverts if the operator is not registered for the application", async () => {
+      await initializeNewFactory()
+      await initializeMemberCandidates()
+
+      await expectRevert(
+        keepFactory.isOperatorUpToDate(members[0], application),
+        "Operator not registered for the application"
+      )
+    })
+  })
+
+  describe("updateOperatorStatus", async () => {
+    let minimumBondableValue
+
+    before(async () => {
+      await initializeNewFactory()
+      await initializeMemberCandidates()
+      await registerMemberCandidates()
+
+      const pool = await FullyBackedSortitionPool.at(signerPool)
+      minimumBondableValue = await pool.getMinimumBondableValue()
+    })
+
+    it("revers if operator is up to date", async () => {
+      await expectRevert(
+        keepFactory.updateOperatorStatus(members[0], application),
+        "Operator already up to date"
+      )
+    })
+
+    it("removes operator if bonding value has changed below minimum", async () => {
+      currentValue = await bonding.unbondedValue(members[0])
+
+      const valueToWithdraw = currentValue.sub(minimumBondableValue).addn(1)
+
+      bonding.withdraw(valueToWithdraw, members[0], {from: members[0]})
+      assert.isFalse(
+        await keepFactory.isOperatorUpToDate(members[0], application),
+        "unexpected status of the operator after bonding value change"
+      )
+
+      await keepFactory.updateOperatorStatus(members[0], application)
+
+      await expectRevert(
+        keepFactory.isOperatorUpToDate(members[0], application),
+        "Operator not registered for the application"
+      )
```

Comment:
We should have another test updating operator status if the value dropped but not removing it from the pool (unbonded value still above the minimum).

Reply:
It's covered by sortition pool test https://github.com/keep-network/sortition-pools/blob/678a7b7d10274329fe77b36fe59a2f34e7fc0f65/test/fullyBackedSortitionPoolTest.js#L293-L323

Reply:
Yeah, but we are testing factory functions here  - we don't need to assert properties of sortition pool but it is worth to cover all `isOperatorUpToDate` and `updateOperatorStatus` paths. The scenario I am writing about is covered in `removes operator if stake has changed below minimum` test of `BondedECDSAKeepFactoryTest`.

Reply:
To summarize:
- `removes operator if bonding value has changed below minimum` - bond dropped below the minimum, operator is removed from the pool as a result of `updateOperatorStatus` and `isOperatorUpToDate` reverts.
- `updates operator if bonding value has changed above minimum` - bond value has changed but the change was insignificant (less than pool's divisor) and pool assumes no change happened so `updateOperatorStatus` reverts.

We do not have the scenario where:
- bonding value dropped, but it is still above the minimum. Operator is updated as a result of `updateOperatorStatus` call and it stays in the pool, so `isOperatorUpToDate` works.
- bonding value changed, it's still above the minimum but the change was significant and update is required. This is covered by `updates operator if stake has changed above minimum` scenario in `BondedECDSAKeepFactoryTest`.

Reply:
Just to clarify the summary: scenarios you mentioned are covered well in sortition-pools, which is where they belong; I'll duplicate them in factory tests as you wish.